_integration-schemas/slack/v1/files.md

```diff
@@ -0,0 +1,245 @@
+---
+tap: "slack"
+version: "1"
+key: ""
+
+name: "files"
+doc-link: "https://api.slack.com/methods/files.list"
+singer-schema: "https://github.com/singer-io/tap-slack/blob/master/tap_slack/schemas/files.json"
+description: |
+  The `{{ table.name }}` table contains {{ integration.display_name }} team files. This list can be filtered with a date window and lookback period in your {{ integration.display_name }} App.
+
+replication-method: "Key-based Incremental"
+
+api-method:
+    name: "files.list"
+    doc-link: "https://api.slack.com/methods/files.list"
+
+attributes:
+  - name: "id"
+    type: "string"
+    primary-key: true
+    description: "The file ID."
+    #foreign-key-id: "file-id"
+
+  - name: "channels"
+    type: "array"
+    description: ""
+    subattributes:
+      - name: "value"
+        type: "string"
+        description: ""
+  - name: "comments_count"
+    type: "integer"
+    description: ""
+  - name: "created"
+    type: "date-time"
+    description: ""
+  - name: "date_delete"
+    type: "date-time"
+    description: ""
+  - name: "display_as_bot"
+    type: "boolean"
+    description: ""
+  - name: "editable"
+    type: "boolean"
+    description: ""
+  - name: "editor"
+    type: "string"
+    description: ""
+  - name: "external_id"
+    type: "string"
+    description: ""
+  - name: "external_type"
+    type: "string"
+    description: ""
+  - name: "external_url"
+    type: "uri"
+    description: ""
+  - name: "filetype"
+    type: "string"
+    description: ""
+  - name: "groups"
+    type: "array"
+    description: ""
+    subattributes:
+      - name: "value"
+        type: "string"
+        description: ""
+  - name: "has_rich_preview"
+    type: "boolean"
+    description: ""
+  
+  - name: "image_exif_rotation"
+    type: "integer"
+    description: ""
+  - name: "ims"
+    type: "array"
+    description: ""
+    subattributes:
+      - name: "value"
+        type: "string"
+        description: ""
+  - name: "is_external"
+    type: "boolean"
+    description: ""
+  - name: "is_public"
+    type: "boolean"
+    description: ""
+  - name: "is_starred"
+    type: "boolean"
+    description: ""
+  - name: "is_tombstoned"
+    type: "boolean"
+    description: ""
+  - name: "last_editor"
+    type: "string"
+    description: ""
+  - name: "mimetype"
+    type: "string"
+    description: ""
+  - name: "mode"
+    type: "string"
+    description: ""
+  - name: "name"
+    type: "string"
+    description: ""
+  - name: "non_owner_editable"
+    type: "boolean"
+    description: ""
+  - name: "num_stars"
+    type: "integer"
+    description: ""
+  - name: "original_h"
+    type: "integer"
+    description: ""
+  - name: "original_w"
+    type: "integer"
+    description: ""
+  - name: "permalink"
+    type: "uri"
+    description: ""
+  - name: "permalink_public"
+    type: "uri"
+    description: ""
+  - name: "pinned_info"
+    type: "object"
+    description: ""
+    subattributes: [
+        ]
```

Comment:
If you see something like this and aren't sure what the subattributes will actually be, just take the whole thing out. So it'll look like this:

```
  - name: "pinned_info"
    type: "object"
    description: ""
```